### ISSUE TITLE
v.2.2.0 has broken the collection of multiple values into a variable called name

### ISSUE THREAD

**Issue by EmilyBourne** [textX/textX#266]
There seems to be a problem with `textx.metamodel.metamodel_from_file` in the new version of textx. I have this error with python 3.5/3.6/3.7/3.8 on linux and mac.

The error I am getting is :
```
            # Special case for 'name' attrib. It is used for cross-referencing
            if hasattr(inst, 'name') and inst.name:
                # Objects of each class are in its own namespace
                if not id(inst.__class__) in parser._instances:
                    parser._instances[id(inst.__class__)] = {}
>               parser._instances[id(inst.__class__)][inst.name] = inst
E               TypeError: unhashable type: 'list'
../../../../virtualenv/python3.8.0/lib/python3.8/site-packages/textx/model.py:512: TypeError
```
It's in the `process_node` function.

The problem seems to be due to the fact that `inst.name` is a list.

I have not yet managed to create a minimum working example. I will submit one when I can

Edit: The problem occurs when calling `textx.metamodel.model_from_str`, not `textx.metamodel.metamodel_from_file` as mentioned above

**Comment by goto40** [comment]
Is it a valid assumption that any attribute called name must be a hashable type, or even a string? In that case we could raise a proper error if this assumption is violated...

**Comment by EmilyBourne** [comment]
It seems that textx is no longer using the `__init__` function of the class. In our `__init__` function we convert the list (non-hashable) to a tuple (hashable)

**Comment by goto40** [comment]
I understand. Thank you very much for the details.

**Comment by goto40** [comment]
Do you use the name attribute to reference the element within your model (scope providers)?

**Comment by EmilyBourne** [comment]
Yes. Here is our model:
```
FunctionMacroStmt:
  'macro' ('(' results*=ID[','] ')' ',')?
  name+=ID['.'] ('(' args*=MacroArg[','] ')')?
  ':='
  master_name+=ID['.'] ('(' master_args*=MacroValue[','] ')')?
;
```

**Comment by goto40** [comment]
Do you have something like `x=[FunctionMacroStmt]` in your grammar?

**Comment by EmilyBourne** [comment]
yes we do. (Here is the file if that helps : https://github.com/pyccel/pyccel/blob/master/pyccel/parser/grammar/headers.tx#L5)

**Comment by EmilyBourne** [comment]
It seems I copied the wrong function name, sorry. The issue is when calling `model_from_str`.

Here is the MWE:

grammat.tx:
```
TestClass:
  'macro' ('(' results*=ID[','] ')' ',')?
  name+=ID['.']
;
```

textx_test.py:
```
from os.path import join, dirname

class TestClass:
    def __init__(self, **kwargs):
        self.name = tuple(kwargs.pop('name'))
        self.results = tuple(kwargs.pop('results'))

this_folder = dirname(__file__)

# Get meta-model from language description
grammar = join(this_folder, 'grammar.tx')

hdr_classes = [TestClass]

from textx.metamodel import metamodel_from_file
meta = metamodel_from_file(grammar, debug=False, classes=hdr_classes)

stmts = 'macro (y), foo.bar'

meta.model_from_str(stmts)
```

**Comment by goto40** [comment]
@EmilyBourne, please correct me: if I did not oversee something, it seems you are not referencing `FunctionMacroStmt`: you have `x=FunctionMacroStmt`, but not `x=[FunctionMacroStmt] `. _In that case_, a simple workaround for you could be to rename the attribute `name` to something else. Then you could use an object processor to restore the name attribute if you need it in your code.

@igordejanovic we should state what properties the name attribute must fulfill. And also under which conditions (always, or only if used for referencing).

**Comment by EmilyBourne** [comment]
You are correct sorry. The simple workaround works 👍 
Thank you for your help

**Comment by igordejanovic** [comment]
> @igordejanovic we should state what properties the name attribute must fulfill. And also under which conditions (always, or only if used for referencing).

@goto40 I agree. There is an issue #40 already registered to provide a better docs and better error report. @EmilyBourne Please see the discussion on the related issue for an another approach to the solution.

**Comment by igordejanovic** [comment]
@EmilyBourne Thanks for the report and MWE. A fix and docs extension is available in PR #269

**Comment by igordejanovic** [comment]
Closed by #269
